torchhd/datasets/collection_datasets.py

```diff
@@ -0,0 +1,403 @@
+            for member in file.getmembers():
+                if member.name.startswith(self.__dataset):
+                    file.extract(member, data_dir)
+
+
+class DatasetFourFold(CollectionDataset):
+    """`Generic class for loading datasets without separate test data that were used in `Do we Need Hundreds of Classifiers to Solve Real World Classification Problems? <https://jmlr.org/papers/v15/delgado14a.html>`_.
+
+    Args:
+        dataset (string): Name of the dataset. Used when downloading
+        root (string): Root directory containing the files of the dataset.
+        train (bool): If True, returns training (sub)set from the file storing training data as further determined by fold and hyper_search variables.
+            Otherwise returns a subset of train dataset if hypersearch is performed (hyper_search = True) if not (hyper_search = False) returns a subset of training dataset
+            as specified in ``conxuntos_kfold.dat`` if fold number is correct. Otherwise issues an error.
+        fold (int): Specifies which fold number to use. The default value of -1 returns the whole train data from the corresponding file.
+            Values between 0 and 3 specify, which fold in ``conxuntos_kfold.dat`` to use. Relevant only if hyper_search is set to False and  0<=fold<=3.
+            Indices in even rows (zero indexing) of ``conxuntos_kfold.dat`` correspond to train subsets while indices in odd rows correspond to test subsets.
+        hyper_search (bool): If True, creates dataset using indeces in ``conxuntos.dat``. This split is used for hyperparameter search. The first row corresponds to train indices (used if train = True)
+            while the second row corresponds to test indices (used if train = False).
+        download (bool): If True, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+        transform (callable): A function/transform that takes in an torch.FloatTensor
+            and returns a transformed version.
+        target_transform (callable): A function/transform that takes in the
+            target and transforms it.
+    """
+
+    # Number of folds for cross validation
+    NUM_FOLDS = 4
+    # Indicates whether the dataset has a separate test file
+    HAS_TEST = False
+
+    def __init__(
+        self,
+        dataset: str,
+        root: str,
+        train: bool,
+        fold: int,
+        hyper_search: bool,
+        download: bool,
+        transform: Optional[Callable],
+        target_transform: Optional[Callable],
+    ):
+        super().__init__(dataset, root, self.HAS_TEST, download)
+        self.__dataset = dataset
+        self.train = train
+        self.fold = fold
+        self.hyper_search = hyper_search
+        self.transform = transform
+        self.target_transform = target_transform
+
+        self._load_data()
+
+    def __len__(self) -> int:
```

Comment:
`__len__` and `__getitem__` seem to be identical between the subclasses so you could hoist it to the parent class.